src/NHibernate/IQueryExpression.cs

```diff
@@ -6,7 +6,7 @@
 namespace NHibernate
 {
 	//TODO 6.0: Merge into IQueryExpression
-	internal interface ICacheableQueryExpression
+	public interface ICacheableQueryExpression
```

Comment:
As explained elsewhere, this interface is just transitional, meant to avoid a binary breaking change in a public interface until v6.0, and should disappear then. So, if the changes int this PR anyway require a v6.0, it is better to merge it in IQueryExpression and get rid of it.

Reply:
OK, i could decouple the ICacheableQueryExpression definition from the ILinqQueryExpression again and revert it back to internal accessibility. But caching of custom ILinqQueryExpression implementations would only work cause all expressions not implementing ICacheableQueryExpression are cached per default. This seems a bit misleading to me. But you mentioned that already earlier.

The changes to copy the plan/expression should also work with the suggested change.

```suggestion
	public interface ILinqQueryExpression: IQueryExpression
	public class NhLinqExpression : ILinqQueryExpression, ICacheableQueryExpression	
```